client/go/outline/client.go

```diff
@@ -49,18 +57,35 @@ type NewClientResult struct {
 	Error  *platerrors.PlatformError
 }
 
+func (c *Client) StartReporting() {
+	if c.ur == nil {
+		return
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	c.cancel = cancel // Store the cancel function to stop reporting later
+
+	go reporting.StartReporting(ctx, c, c.ur)
+}
+
+func (c *Client) StopReporting() {
+	if c.cancel != nil {
+		c.cancel() // Signal the context to stop reporting
+		c.cancel = nil
+	}
+}
+
 // NewClient creates a new Outline client from a configuration string.
-func NewClient(transportConfig string) *NewClientResult {
+func NewClient(transportConfig string, sessionConfig string) *NewClientResult {
```

Comment:
This is not a session config. It's a report config. But we should merge them into a client config.